gradle-consistent-versions-idea-plugin/src/main/java/com/palantir/gradle/versions/intellij/RepositoryExplorer.java

```diff
@@ -103,21 +108,39 @@ private Set<GroupPartOrPackageName> fetchFoldersFromContent(String contents) {
     }
 
     private Set<DependencyVersion> parseVersionsFromContent(String content) {
-        Set<DependencyVersion> versions = new HashSet<>();
         try {
             XmlMapper xmlMapper = new XmlMapper();
+            xmlMapper.registerModule(new GuavaModule());
 
             Metadata metadata = xmlMapper.readValue(content, Metadata.class);
-            if (metadata.versioning() != null && metadata.versioning().versions() != null) {
-                String latest = metadata.versioning().latest();
-                for (String version : metadata.versioning().versions()) {
-                    versions.add(DependencyVersion.of(version, latest.equals(version)));
-                }
+            List<String> allVersions = new ArrayList<>(metadata.versioning().versions());
+            Collections.reverse(allVersions);
+
+            if (allVersions.isEmpty()) {
+                return Collections.emptySet();
+            }
+
+            String latest = Optional.ofNullable(metadata.versioning().release())
+                    .filter(l -> !l.isEmpty())
+                    .orElseGet(() -> metadata.versioning().latest());
+
+            if (latest.toLowerCase(Locale.ROOT).contains("rc")
+                    || latest.toLowerCase(Locale.ROOT).contains("snapshot")) {
+                latest = allVersions.stream()
+                        .filter(version -> !version.toLowerCase(Locale.ROOT).contains("rc")
+                                && !version.toLowerCase(Locale.ROOT).contains("snapshot"))
```

Comment:
I'd check they end with `-rc*` or `-SNAPSHOT` (I could totally see an artifact having `rc` or `snapshot` in the name). Could probably use regex/Pattern. I think we'd want `-M1`, `-M2` as well as `-alpha` and `-beta`

Reply:
Oh wait, this is just checking the version. Probably still best to just check they end with.

Reply:
fixed - now using regex match and looking for a wider range of matches